packages/material/src/complex/MaterialTableControl.tsx

```diff
@@ -116,7 +116,7 @@ const generateCells = (
 };
 
 const getValidColumnProps = (scopedSchema: JsonSchema) => {
-  if (scopedSchema.type === 'object') {
+  if (scopedSchema.type === 'object' && typeof scopedSchema.properties === 'object') {
```

Comment:
shouldn't a check for not undefined be enough?

Reply:
We're using `Object.keys(scopedSchema.properties)` later which expects an object-type parameter. Depending on the JS version it will either throw a `TypeError` or does a meaningless coercion on non-object types. So we really should check for `object`.